### PR TITLE
3unks the Threequel

### PR DESCRIPTION
**Pull request by ethteck**
They say there's nothing better than the first of a series ( #920 ), and sequels ( #952 ) rarely live up to the original. But the studio just didn't want to let our story die here. I present 3unks the third!

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀 This PR matches 3 functions (+0.09%) on `us`.
🚀 This PR matches 1 function (+0.02%) on `jp`.
🚀 This PR matches 1 function (+0.02%) on `cn`.

**Comment by BowserSlug** [comment]
🚀🚀🚀 This PR matches 3 functions (+0.09%) on `us`.
🚀 This PR matches 1 function (+0.02%) on `jp`.
🚀 This PR matches 1 function (+0.02%) on `cn`.

File: src/effects/effect_65.c [hunk 1]
```diff
@@ -257,124 +258,134 @@ void fx_65_appendGfx(void* effect) {
...
     }
-    var_fp = 0;
-    sp60 = -1;
-    sp64 = (sp50 & 0x3F) << 5;
-    gDPSetPrimColor(gMasterGfxPos++, 0, 0, data->unk_18, data->unk_1C, data->unk_20, temp_s5);
+
+    gDPSetPrimColor(gMasterGfxPos++, 0, 0, data->unk_18, data->unk_1C, data->unk_20, primAlpha);
     gDPSetEnvColor(gMasterGfxPos++, data->unk_28, data->unk_2C, data->unk_30, 0);
-    gSPBranchList(gMasterGfxPos, &gMasterGfxPos[121]);
+    gSPBranchList(gMasterGfxPos, gMasterGfxPos + 0x79);
```
Comment (HailSanta) [inline src/effects/effect_65.c:264]: Might be nice to express this 0x79 in terms on which gfx commands are being jumped over. Seems like 29x gSPVertex/gSP2Triangles + gSPPopMatrix + ???

File: include/variables.h [hunk 2]
```diff
@@ -103,7 +103,7 @@ extern s16 D_800A095E;
 extern u8 D_800A095F;
 extern s32 D_800A0960;
 extern s32 D_800A0964;
-extern s8 D_800A0B94;
+extern s8 gNpcCollisionsEnabled;
```
Comment (HailSanta) [inline include/variables.h:106]: Maybe specify gNpcPlayerCollisionsEnabled? I don't think this affects NPC-NPC collisions.

File: src/npc.c [hunk 3]
```diff
@@ -446,11 +442,103 @@ void npc_do_other_npc_collision(Npc* npc) {
...
+            wPartnerNpc->pos.x -= temp_f24;
+            wPartnerNpc->pos.z -= playerZ;
+        } else {
+            playerStatus->position.x -= temp_f24 * 0.5f;
+            playerStatus->position.z -= playerZ * 0.5f;
+            wPartnerNpc->pos.x -= temp_f24 * 0.5f;
+            wPartnerNpc->pos.z -= playerZ * 0.5f;
+        }
+    } else {
+        if (playerStatus->flags & (4 | 2)) {
```
Comment (HailSanta) [inline src/npc.c:523]: PS_FLAG_JUMPING | PS_FLAG_FALLING

File: src/world/script_api/push_blocks.c [hunk 4]
```diff
@@ -188,15 +188,14 @@ API_CALLABLE(func_80282E30) {
     script->varTable[4] = yThing = entity->position.y;
     script->varTable[5] = entityZ = entity->position.z;
 
-    xThing = (xThing - grid->centerPos.x);
+    xThing = xThing - grid->centerPos.x;
+    entityZ = entityZ - grid->centerPos.z;
```
Comment (HailSanta) [inline src/world/script_api/push_blocks.c:192]: The previous steps here seem cleaner to me. Was this necessary to keep the match?

**Review by HailSanta — APPROVED** [review]

File: src/world/script_api/push_blocks.c (hunk 4 above)
Comment (ethteck) [reply src/world/script_api/push_blocks.c:192]: it's not a match yet - just some WIP progress on the function

**Pull request merged**